### PR TITLE
[Hub Generated] Review request for Microsoft.Capacity to add version preview/2019-04-01

### PR DESCRIPTION
**Pull request by Only2125**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Only2125/azure-rest-api-specs/dev-reservations-Microsoft.Capacity-2019-04-01...master/).

### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 1]
```diff
@@ -27,6 +27,50 @@
     }
   },
   "paths": {
+    "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/reservations/{reservationId}/availablescopes": {
+      "post": {
+        "summary": "Get Avaialble Scopes for `Reservation`.",
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]:
```suggestion
        "summary": "Get Available Scopes for `Reservation`.",
```

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 2]
```diff
@@ -27,6 +27,50 @@
...
+        "description": "Get Avaialble Scopes for `Reservation`.\n",
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]:
```suggestion
        "description": "Get Available Scopes for `Reservation`.\n",
```

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 3]
```diff
@@ -27,6 +27,50 @@
...
+        "operationId": "Reservation_availablescopes",
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]:
```suggestion
        "operationId": "Reservation_AvailableScopes", 
```

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 4]
```diff
@@ -27,6 +27,50 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          }
+        ],
+        "tags": [
+          "Reservation, availablescopes"
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]:
```suggestion
          "Reservation, AvailableScopes"
```

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 5]
```diff
@@ -27,6 +27,50 @@
     }
   },
   "paths": {
+    "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/reservations/{reservationId}/availablescopes": {
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]:
```suggestion
    "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/reservations/{reservationId}/availableScopes": {
```

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json [hunk 6]
```diff
@@ -27,6 +27,50 @@
...
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          }
+        ],
+        "tags": [
+          "Reservation, AvailableScopes"
```
Comment (nschonni) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]: Note sure if these are supposed to be separate tags
```suggestion
          "Reservation", "AvailableScopes"
```

Comment (Only2125) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]: Not sure, but I checked the format of our previous swagger. Same as what I put.

Comment (Only2125) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]: Let me know if I need to change this format.

Comment (nschonni) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2019-04-01/reservations.json]: Sorry, one of the SDK/ARM folks would need to make that call. I just noticed that it looked a little odd

**Comment by ravbhatnagar** [comment]
Signing off from ARM side (new POST /availableScopes API).

**Review by yaohaizh — APPROVED** [review]

**Review by yaohaizh — APPROVED** [review]

**Pull request merged**